execute/source_iterator.go

```diff
@@ -85,3 +85,93 @@ func (c *sourceIterator) Run(ctx context.Context) {
 		t.Finish(c.id, err)
 	}
 }
+
+// CreateSourceFromIterator takes an implementation of a SourceIterator as well as a dataset ID
+// and creates an execute.Source.
+func CreateSourceFromIterator(iterator SourceIterator, dsid DatasetID) (Source, error) {
+	return &sourceIterator2{iterator: iterator, id: dsid}, nil
```

Comment:
I think it's confusing that neither `sourceIterator` nor `sourceIterator2`  implement the interface `SourceIterator`.   What if we rename `SourceIterator`  to `TableIterator`  ?    

And then we can also rename `sourceIterator`  to `decoderSource`  and `sourceIterator2` to `tableIteratorSource`.  

Long-term, we should eliminate the tech-debt and keep just the TableIterator types.

Reply:
I was going to rename these but I just kind of forgot.

Reply:
Ah, I didn't understand this message. I think that's probably a good idea to just have this implement `TableIterator` and make a default source that consumes a `TableIterator`.

Reply:
I modified `SourceIterator` to use the same signature but added `context.Context`. The `TableIterator` doesn't take a context, but the sources need the context so they know when to cancel.

So the interfaces are functionally the same, but slightly different because of the context parameter. It worked a lot better and was much easier to implement.